### PR TITLE
Apiresult

### PR DESCRIPTION
**Pull request by Shweta4321**
**Checklist**

- [x ] My branch is up-to-date with the upstream `develop` branch.

**Which issue does this PR fix?**: 
fixes: #95 

**what this PR does?**
we can search our query in wikipedia  or stackoverfolw

**Why do we need this PR?**:
it gives relevant answer

If relevant, please include a screenshot.
screenshot from 2018-03-17 15-12-25
screenshot from 2018-03-17 15-12-54
screenshot from 2018-03-17 15-13-01

### CODE REVIEW

File: search/templates/cosmos/searchresults.html [hunk 1]
```diff
@@ -50,6 +50,31 @@ <h4 class="card-title">
...
+                     <a href="#tab2default" style="color: #1b1e21" data-toggle="tab"><img src="../static/images/stack-overflow.png"/></a>
+</li>
+</ul>
+
+                    </div>
+                    <div class="tab-content">
+                        <div class="tab-pane fade in active" id="tab1default">
+                            <p><a href="{{ res_wiki.url }}" target="blank"/>{{ res_wiki.heading }}</p>
+                        </div>
+                        <div class="tab-pane fade" id="tab2default">
```
Comment (iattempt) [inline search/templates/cosmos/searchresults.html]: The ids of those dropdown need be difference.

File: search/templates/cosmos/searchresults.html [hunk 2]
```diff
@@ -50,6 +50,31 @@ <h4 class="card-title">
                                 aria-haspopup="true" aria-expanded="false">
                             Solutions
                         </button>
+                    <div>
+                    <div >
+                        <ul class="sol_tab nav nav-tabs">
+                            <li class="active">
+                     <a href= "#tab1default" style="color: #1b1e21" data-toggle="tab">
+                         <img src="../static/images/wikipedia.png"/>
+                     </a>
```
Comment (iattempt) [inline search/templates/cosmos/searchresults.html]: Please check indentations.

**Comment by Shweta4321** [comment]
@AdiChat  please review .

File: search/templates/cosmos/searchresults.html (hunk 1 above)
Comment (Shweta4321) [reply search/templates/cosmos/searchresults.html]: Will you please tell about which dropdown you are talking, because as of now there is no dropdown in that script.

Comment (iattempt) [reply search/templates/cosmos/searchresults.html]: Sorry, what I want to say is `collapse`, you could try to clicking wiki/stack on second result, then you get it.

**Comment by AdiChat** [comment]
Nice work :+1:

You need to include the images in this pull request. You may place them beside the `Solutions` button.

You may include this Wikipedia and StackOverflow results for each result from Cosmos. If you search `search`, several search algorithms from Cosmos comes up and the Wikipedia and StackOverflow icons are present in each result, though, clicking on them opens result in the first result only. You are searching for the search term but for this path, you must take the Cosmos absolute path of the result (example: `binary search` for the first result for `search`) and use this as a search term for the APIs. 

A user may not explore all Wikipedia and StackOverflow results for all Cosmos results, so, you may use AJAX and Lazy loading. 

This will improve the user learning experience as it will give the opportunity to the user to explore a subsection deeper. 

Kindly take a look into this. :+1:

**Comment by Shweta4321** [comment]
@AdiChat  Review.

**Comment by Shweta4321** [comment]
@AdiChat  Review

**Comment by AdiChat** [comment]
Great work :+1: 

You need to include the images in this pull request. 

Consider the results for the term "search". The Wikipedia results are fine but the StackOverflow results are the same for every result indicating that it is for the keyword "search" not for a particular result like "linear search". 

The UI is broken. Lower results are nested in upper results. See for the search term "search". 

Kindly take a look at this. :+1:

**Comment by Shweta4321** [comment]
@AdiChat  In  case of more numbers of Solutions  like in ```Sort ```, it is giving  ```too many requests from this IP``` error

screenshot from 2018-04-07 19-27-21

**Comment by AdiChat** [comment]
Yes, this issue will remain even if we resolve it for this case by using AJAX as there will be multiple users in production. 

One solution is to write a script to search a search engine like Google for keywords like ` site:stackoverflow.com` and ` site:*.stackexchange.com`, collect and store the results in a database ( like [Solr](http://django-haystack.readthedocs.io/en/master/installing_search_engines.html)) and use it to serve results through Cosmos-Search.
In short, we need to create an equivalent of its sitemap as it is not publically available. 

This can be done for Wikipedia as well. 

Once done, we can update the database in a couple of months to keep new results in consideration.

**Comment by Shweta4321** [comment]
@AdiChat  Please Review

**Comment by AdiChat** [comment]
Nice work :+1: 

The UI and functionality look perfect. 

As all search requests will be made from a static IP and as the number of users will increase, this feature state may break as requests to Wikipedia API and Google search will be restricted. 

A more robust solution is to crawl selected sites and use the stored data to present results. A couple of crawler code in Python are available which can be modified for our purpose. 

Kindly take a look into this. :+1:

**Comment by Shweta4321** [comment]
@AdiChat  please review
